b2/console_tool.py

```diff
@@ -2598,43 +2608,227 @@ def run(self, args):
         return 0
 
 
-@B2.register_subcommand
-class UploadFile(
-    DestinationSseMixin, LegalHoldMixin, FileRetentionSettingMixin, UploadModeMixin, Command
+class UploadFileMixin(
+    ProgressMixin, DestinationSseMixin, LegalHoldMixin, FileRetentionSettingMixin
 ):
     """
-    Uploads one file to the given bucket.  Uploads the contents
-    of the local file, and assigns the given name to the B2 file,
+    Content type is optional.
+    If not set, it will be guessed.
+
+    The maximum number of upload threads to use to upload parts of a large file is specified by ``--threads``.
+    It has no effect on "small" files (under 200MB as of writing this).
+    Default is 10.
+
+    Each fileInfo is of the form ``a=b``.
+    """
+
+    @classmethod
+    def _setup_parser(cls, parser):
+        parser.add_argument(
+            '--quiet', action='store_true', help="prevents printing any information to stdout"
+        )
+        parser.add_argument(
+            '--contentType',
+            help="MIME type of the file being uploaded. If not set it will be guessed."
+        )
+        parser.add_argument(
+            '--minPartSize', type=int, help="minimum size of a part for multipart uploads"
+        )
+        parser.add_argument(
+            '--sha1', help="SHA-1 of the data being uploaded for verifying file integrity"
+        )
+        parser.add_argument(
+            '--threads', type=int, default=10, help="number of threads used for the operation"
+        )
+        parser.add_argument('--cache-control', default=None)
+        parser.add_argument(
+            '--info',
+            action='append',
+            default=[],
```

Comment:
the default being mutable can backfire in some strange corner case, no? @mjurbanski-reef